### DISCUSSION TITLE
About the Possibility of Performing Time- and Frequency-Resolved (Spectral) MVPA using MVPAlab

### DISCUSSION THREAD

**Discussion by batiyilmaz** [dlopezg/mvpalab#17] [Q&A] [answered]
Dear David,

I just wanted to kindly ask whether it is possible to perform Time- and frequency-resolved MVPA using the toolbox. To my understanding MVPAlab requires input.data, input.times, input.fs, and input.format as the input, hence using [electrodes x timepoints x trials]  as a feature. Would it be possible to modify the custom data, time-frequency decompose it and use [electrodes x timepoints x trials x frequency] as a feature in a time and frequency resolved manner, thus performing spectral MVPA for each timepoint and frequency by modifying the input.data accordingly? In other words, instead of using raw value activation patterns (i.e., raw voltage) as features, using power value patterns as features in this fashion?

On another note, from the reference paper of MVPAlab, I noticed this statement: "...As mentioned, MVPAlab Toolbox was initially designed to work with epoched M/EEG data, extracting the raw potential of the signal and computing time-resolved classification analyses. The latest release supports different signal characteristics as features, such as the power envelope or the instantaneous phase of the signal."

Does the latest release support instantaneous phase of the signal as a feature as well? 

Thank you so much for your time and immense effort. I'm looking forward to hearing about this.

Best,
Bati

**Comment by dlopezg** [comment]
Dear Bati,

Unfortunately, at the moment it is not possible to directly compute TF resolved MVPA in the toolbox. However, I guess you could try to previously decompose your time resolved data in different frequency bands, and use them as inputs of regular time-resolved MVPA analyses. It shouldn't be difficult to concatenate the obtained results and plot them in a n_freq x n_timepoints matrix. 

The instantaneous phase of the signal was not finally implemented in the released version of the toolbox.

Best,
David.

**Reply by batiyilmaz** [reply]
Dear David,

Thank you for your reply, I appreciate it. Can I also kindly ask that when previously decompose the raw EEG data in that manner and use it as a feature for the time-resolved MVPA, would the classifier iterate over each individual frequency value like in the case of iterating over each timepoint when specified?

Lastly, would it be possible to perform TG-MVPA and TG-MVCC in this fashion? Would that need additional requirements? 

I would greatly appreciate your insights into all of this.

Thank you so much for your time and help in advance.

Best,
Bati

**Reply by dlopezg** [reply, marked as answer]
Dear Bati,

Unfortunately, this feature has not been implemented. The classifier currently iterates over each time point but not across different frequency bands. To achieve the desired behavior, you should decompose your data beforehand. Then, you need to create a loop for each frequency band of interest and compute the regular MVPA analysis for each iteration. Save the obtained results in different files and represent them in the same plot at the end.

I understand that this is not straightforward solution, and coding is necessary to obtain the desired results, but it's the best solution that comes to my mind at the moment.

Alternatively, you could follow a similar approach for computing the TG analysis for a specific frequency band. First, extract the power of the signal in the desired band (using EEGLAB or another toolbox), save different datasets for different frequency ranges of interest, and use them as input for a regular TG analysis.

We would like to implement these and other new features at some point, but I am unable to provide you with a specific date.

Best regards,
David

**Reply by batiyilmaz** [reply]
Dear David, 

Thank you so much for your detailed answer. I'm hoping to follow up this potential solution by manually decomposing and creating a loop for each frequency band. 

While I'm working on this, in the meantime, would it be possible to kindly learn about the potential date of the next release enabling sliding filter analysis (frequency contribution analysis) running without any errors?

Thank you so much for your help and answers.

Best,
Bati

**Reply by dlopezg** [reply]
Hi Bati,

I am trying to replicate and solve the error so the next release will be ready as soon.

Best, 
David

**Reply by batiyilmaz** [reply]
Hi David,

Thank you for your reply. As I appreciate your effort in the toolbox, if in anyway, I could be helpful with my suggestions and reports, I would like to help further as well. Please let me know.

For example, I have another question regarding the latency statistics of time series (i.e., obtaining the onset, offset, and peak latency values) while computing the cluster-based permutation + bootstrapping technique via the toolbox but I can ask it later as you are busy at the moment.

Looking forward to downloading and using the next release.

Thank you again.

Best,
Bati

**Reply by batiyilmaz** [reply]
Hi David,

I wanted to kindly ask about whether there is any update for the Sliding Filter Analysis (Frequency Contribution Analysis), I have been using the toolbox in the time-resolved way for a while now and I would like to thank you again for your contributions! I'm looking forward to experimenting with the SF-Analysis! Thanks!

**Reply by dlopezg** [reply]
Dear Bati,

I have been quite busy these past few weeks, and the updated version of the toolbox, fixing different bugs, has not been released yet. I will be launching the new version very soon. Thank you for your patience!

David

**Reply by batiyilmaz** [reply]
Thank you for your answer, David!

**Comment by batiyilmaz** [comment]
Hi David, 

Since I finished the decomposition, I will follow the procedure described here now, the data that I have has 30 frequencies, from 1-30 Hz. I wanted to ask will I loop through each of them and then compute the regular MVPA by using the run_mvpa.m? And then will I able to plot it as usual using the MVPAlab plotting functions?

Thanks,
Bati

**Reply by batiyilmaz** [reply]
Finally, will the statistical methods used in MVPAlab work as usual while plotting this time-frequency-resolved MVPA in a TGM format?
